clang/include/clang/StaticAnalyzer/Core/AnalyzerOptions.def

```diff
@@ -299,13 +299,12 @@ ANALYZER_OPTION(
 
 ANALYZER_OPTION(
     bool, ShouldEagerlyAssume, "eagerly-assume",
-    "Whether we should eagerly assume evaluations of conditionals, thus, "
-    "bifurcating the path. This indicates how the engine should handle "
-    "expressions such as: 'x = (y != 0)'. When this is true then the "
-    "subexpression 'y != 0' will be eagerly assumed to be true or false, thus "
-    "evaluating it to the integers 0 or 1 respectively. The upside is that "
-    "this can increase analysis precision until we have a better way to lazily "
-    "evaluate such logic. The downside is that it eagerly bifurcates paths.",
+    "If this is enabled (the default behavior), when the analyzer encounters "
+    "a comparison operator or logical negation, it immediately splits the "
```

Comment:
Good point, if you implement it in a follow-up non-NFC commit, I'd be happy to review it. (Alternatively, we could create a todo note / github ticket etc. to record this.)

Reply:
I don't get your response here. I thought we want to update the doc comments here to reflect what the function does. I raised that I think implicit conversions also trigger this code path (and do eager splits). Consequently, I figured that this behavior should be also mentioned here and now.

If bool conversions don't trigger this function, then of course my suggestion is void.
I didn't mean to suggest function changes by my comment.

About the TODO notes: I wish we had some lightweight list of things that we as maintainers agree that is easy to do, we have a clear NFC or semantic change in mind but nobody has the time doing it. I have in mind a lot of NFC refactors, API cleanups where I never get to, but I wish would be done.

Reply:
Oh, I see, I misunderstood your comment. 

Currently implicit bool conversions _do not_ trigger the eagerly-assume code path, and I automatically assumed that you know this and your suggestion says that we should modify this behavior.

In fact I still think that it might be worthwhile to implement this in a follow-up commit (although I don't think that I'll work on this soon); but then it's clear that there is nothing to do with this PR at this point.

> About the TODO notes: I wish we had some lightweight list of things that we as maintainers agree that is easy to do, we have a clear NFC or semantic change in mind but nobody has the time doing it. I have in mind a lot of NFC refactors, API cleanups where I never get to, but I wish would be done.

Nowdays I tend to record these ideas for myself by creating tickets in our internal system; but perhaps I should switch to creating github issues because that would be visible for other contributors as well. Creating a github ticket is not as lightweight as e.g. writing a comment, but I think that it'd be still acceptable. Perhaps we should create a tag/group/whatever for this kind of issue to be able to list them easily.

Reply:
Thanks for the explanation.